components/email/useEmailRecipientStatus.tsx

```diff
@@ -0,0 +1,33 @@
+import { DOMAINS_ALLOWING_EMAIL_COMMUNICATION } from '@/lib/constants'
+import { resolveDidDocData } from '@/lib/identity'
+import { useQuery } from '@tanstack/react-query'
+
+export const useEmailRecipientStatus = (
+  did: string,
+): { isLoading: boolean; error: any; cantReceive: boolean } => {
+  const { data, isLoading, error } = useQuery({
+    queryKey: ['email-capability-check', did],
+    queryFn: async () => {
+      const response = await resolveDidDocData(did)
+      if (!response?.services) {
+        return false
+      }
+
+      const pdsEndpoint = response.services['atproto_pds']?.endpoint
+      if (!pdsEndpoint) {
+        return false
+      }
+
+      return DOMAINS_ALLOWING_EMAIL_COMMUNICATION.some((domain) => {
+        return pdsEndpoint.endsWith(domain)
```

Comment:
We probably want the domain to end with `.bsky.network` rather than `bsky.network`.

Reply:
makes sense. but since this is env var, we can just configure it that way in the var, right? maybe the name needs an adjustment so that it's not DOMAINS_ALLOWING_EMAIL_COMMUNICATION and instead DOMAINS_SUFFIXES_ALLOWING_EMAIL_COMMUNICATION?

Reply:
@devinivy do lmk if you have thoughts on this, I'll merge for now and happy to address any different outcome from this as a fast follow.

Reply:
Yup we can just configure it that way 👍